manifests/env.pp

```diff
@@ -6,21 +6,21 @@
 # parameter
 
 define poudriere::env (
-  $ensure           = 'present',
-  $makeopts         = [],
-  $makefile         = nil,
-  $version          = '10.0-RELEASE',
-  $arch             = 'amd64',
-  $jail             = $name,
-  $paralleljobs     = $facts['processors']['count'],
-  $pkgs             = [],
-  $pkg_file         = nil,
-  $pkg_makeopts     = {},
-  $pkg_optsdir      = nil,
-  $portstree        = 'default',
-  $cron_enable      = false,
-  $cron_always_mail = false,
-  $cron_interval    = { minute => 0, hour => 0, monthday => '*', month => '*', weekday => '*' },
+  Enum['present', 'absent'] $ensure           = 'present',
+  Array[String[1]]          $makeopts         = [],
+  Optional[String::Abspath] $makefile         = undef,
+  String[1]                 $version          = '10.0-RELEASE',
+  String[1]                 $arch             = 'amd64',
```

Comment:
this is a known set that could also be an enum, but we can move this down to the better defaults refactoring

Reply:
That list changed a few times in the last years… I think `sparc64` is not an option anymore, `arm` and `arm64` where added more recently.  I am not sure if the `Enum` is the best in this case, but I do not have strong opinions about it.

Reply:
and this module might not be the best place for it, maybe something closer to the FreeBSD system could carry that info / type